src/components/hoc/with-blog-service.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { BlogServiceConsumer } from '../blog-service-context';
+
+const withBlogService = () => Wrapped => {
```

Comment:
No need in HOC and in context, it is overachitecture - you should send API calls using redux side-effects.